### PR TITLE
New Packages: kmail and deps

### PR DESCRIPTION
**Pull request by Johnnynator**
Only tested on x86_64 so far.
TODO:
- [x] test cross-compiling
- [x] ~~test musl~~ segfaults
- [x] mysql dependecy for akonadi or kmail ?!

### CODE REVIEW

File: srcpkgs/akonadi-notes/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'akonadi-notes'
+pkgname=akonadi-notes
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules python"
+makedepends="kmime-devel akonadi5-devel"
+short_desc="Libraries and daemons to implement notes management in Akonadi"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-2.1-or-later GPL-2-or-later BSD-2-Clause"
```
Comment (maxice8) [inline srcpkgs/akonadi-notes/template]: comma separate licenses, as per the holy manual.md

> license A string matching any license file available in /usr/share/licenses. Multiple licenses should be separated by commas, i.e GPL-3, LGPL-2.1.

also GPL-2-or-later isn't valid SPDX, GPL-2.0-or-later is. Thanks for your efforts on adopting SPDX

File: srcpkgs/akonadi-search/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+pkgname=akonadi-search
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="akonadi-mime-devel xapian-core-devel krunner-devel kcmutils-devel
+ kcontacts-devel kcalcore-devel"
+short_desc="Libraries and daemons to implement searching in Akonadi"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/akonadi-search/template]: needs SPDX identifier.

File: srcpkgs/calendarsupport/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'calendarsupport'
+pkgname=calendarsupport
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-tools python3"
+makedepends="kholidays-devel akonadi-calendar-devel kdepim-apps-libs-devel"
+short_desc="Calendar support library"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later LGPL-2.1.-or-later"
```
Comment (maxice8) [inline srcpkgs/calendarsupport/template]: GPL-2 not being SPDX likewise, see comment on `akonadi-notes`

File: srcpkgs/extra-cmake-modules/template [hunk 4]
```diff
@@ -12,7 +12,7 @@ maintainer="Juan RP <xtraeme@voidlinux.eu>"
 license="BSD"
```
Comment (maxice8) [inline srcpkgs/extra-cmake-modules/template]: Would be nice if it is updated to reflect the actual license it uses, BSD gotta be the most vague identifier that exists.

File: srcpkgs/grantleetheme/template [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+pkgname=grantleetheme
+version=17.12.3
+revision=1
+build_style=cmake
+configure_args="KDE_INSTALL_USE_QT_SYS_PATHS=TRUE"
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="grantlee5-devel ki18n-devel kiconthemes-devel knewstuff-devel"
+short_desc="Library for Grantlee theming support"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/grantleetheme/template]: not SPDX identifier.

File: srcpkgs/kcalcore/template [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'kcalcore'
+pkgname=kcalcore
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python bison"
+makedepends="libical-devel qt5-devel"
+short_desc="The KDE calendar access library"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/kcalcore/template]: not a valid SPDX identifier

File: srcpkgs/kdav/template [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'kdav'
+pkgname=kdav
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools"
+makedepends="kio-devel qt5-xmlpatterns-devel"
+short_desc="DAV protocol implemention with KJobs"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later"
```
Comment (maxice8) [inline srcpkgs/kdav/template]: not valid SPDX

File: srcpkgs/kdepim-apps-libs/template [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+pkgname=kdepim-apps-libs
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="grantlee5-devel kcontacts-devel akonadi-contacts-devel libkleo-devel
+ grantleetheme-devel libkdepim-devel pimcommon-devel"
+short_desc="KDE PIM mail related libraries"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/kdepim-apps-libs/template]: not valid SPDX identifier.

File: srcpkgs/kdepim-runtime/template [hunk 9]
```diff
@@ -1,17 +1,19 @@
...
-makedepends="libressl-devel qt-devel phonon-devel kdelibs-devel
- kdepimlibs-devel akonadi-devel boost-devel libxml2-devel
- qjson-devel libkgapi-devel"
+hostmakedepends="extra-cmake-modules python3 kdoctools kdesignerplugin"
+makedepends="akonadi-calendar-devel akonadi-notes-devel knotifyconfig-devel
+ pimcommon-devel kdav-devel libkgapi-devel kalarmcal-devel kmbox-devel"
 short_desc="KDE PIM runtime applications/libraries"
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
-license="GPL, LGPL, FDL"
+license="GPL-2-or-later GPL-3-or-later LGPL-2.1-or-later AGPL-3.0-or-later"
```
Comment (maxice8) [inline srcpkgs/kdepim-runtime/template]: commas separate licenses.

File: srcpkgs/kimap/template [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'kimap'
+pkgname=kimap
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="kio-devel kmime-devel"
+short_desc="Job-based API for interacting with IMAP servers"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/kimap/template]: not valid SPDX.

File: srcpkgs/kldap/template [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+pkgname=kldap
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python
+ kdoctools"
+makedepends="kio-devel"
+short_desc="LDAP access API for KDE"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/kldap/template]: not valid SPDX

File: srcpkgs/kmail-account-wizard/template [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+pkgname=kmail-account-wizard
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="libkleo-devel kross-devel kmailtransport-devel kidentitymanagement-devel
+ pimcommon-devel knotifyconfig-devel ktexteditor-devel"
+short_desc="KMail account wizard"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later"
```
Comment (maxice8) [inline srcpkgs/kmail-account-wizard/template]: not valid SPDX

File: srcpkgs/kmail/template [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules python qt5-host-tools qt5-qmake"
+makedepends="kcalutils-devel knotifyconfig-devel kontactinterface-devel
+ kparts-devel ktnef-devel libkleo-devel libksieve-devel mailcommon-devel
+ messagelib-devel qt5-location-devel qt5-webchannel-devel qt5-webengine-devel"
+depends="kdepim-runtime kmail-account-wizard"
+short_desc="KDE Mail Client"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later LGPL-2.1-or-later GFDL-1.2-or-later"
```
Comment (maxice8) [inline srcpkgs/kmail/template]: not valid SPDX on GPL-2-or-later and comma separates the licenses.

File: srcpkgs/kmailtransport/template [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'kmailtransport'
+pkgname=kmailtransport
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python kconfig"
+makedepends="akonadi-mime-devel kcmutils-devel ksmtp-devel"
+short_desc="Mail Transport Service"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/kmailtransport/template]: not valid SPDX

File: srcpkgs/ksmtp/template [hunk 15]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'ksmtp'
+pkgname=ksmtp
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="kmime-devel kio-devel"
+short_desc="Job-based library to send email through an SMTP server"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/ksmtp/template]: not valid SPDX

File: srcpkgs/ktnef/template [hunk 16]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'ktnef'
+pkgname=ktnef
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="kcalutils-devel kcontacts-devel"
+short_desc="API for handling TNEF data"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/ktnef/template]: not valid SPDX

File: srcpkgs/libgravatar/template [hunk 17]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'libgravatar'
+pkgname=libgravatar
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="pimcommon-devel"
+short_desc="KDE PIM library providing Gravatar support"
+maintainer="John <johnz@posteo.net>"
+license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/libgravatar/template]: not valid SPDX

File: srcpkgs/libkdepim/template [hunk 18]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'libkdepim'
+pkgname=libkdepim
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="akonadi-contacts-devel kcmutils-devel kldap-devel akonadi-search-devel"
+short_desc="Libraries for KDE PIM applications"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/libkdepim/template]: not valid SPDX

File: srcpkgs/libkleo/template [hunk 19]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'libkleo'
+pkgname=libkleo
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules python qt5-qmake qt5-host-tools"
+makedepends="kitemmodels-devel kpimtextedit-devel gpgmeqt-devel gpgmepp-devel"
+short_desc="KDE PIM cryptographic library"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/libkleo/template]: not valid SPDX

File: srcpkgs/libksieve/template [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+pkgname=libksieve
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python kdoctools"
+makedepends="kidentitymanagement-devel kmailtransport-devel pimcommon-devel
+ qt5-webengine-devel qt5-webchannel-devel qt5-location-devel"
+short_desc="KDE PIM library for managing sieves"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later LGPL-2.1-or-later"
```
Comment (maxice8) [inline srcpkgs/libksieve/template]: not valid SPDX and commas separate the licenses.

File: srcpkgs/mailcommon/template [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+pkgname=mailcommon
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="mailimporter-devel messagelib-devel kdesignerplugin-devel
+ pimcommon-devel libkleo-devel"
+short_desc="KDE PIM library providing support for mail applications"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later LGPL-2.1-or-later"
```
Comment (maxice8) [inline srcpkgs/mailcommon/template]: not valid SPDX and comma separate the licenses.

File: srcpkgs/mailimporter/template [hunk 22]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'mailimporter'
+pkgname=mailimporter
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="libkdepim-devel"
+short_desc="Mail importer library"
+maintainer="John <johnz@posteo.net>"
+license="GPL-2-or-later LGPL-2.1-or-later"
```
Comment (maxice8) [inline srcpkgs/mailimporter/template]: not valid SPDX and liceses are delimitated by commas.

File: srcpkgs/messagelib/template [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="akonadi-contacts-devel gpgmeqt-devel kdepim-apps-libs
+ kidentitymanagement-devel kmailtransport-devel kmbox-devel libgravatar-devel
+ libkdepim-devel qt5-location-devel qt5-webchannel-devel qt5-webengine-devel"
+short_desc="KDE PIM messaging library"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/messagelib/template]: not valid SPDX identifier.

File: srcpkgs/pimcommon/template [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+version=17.12.3
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules qt5-qmake qt5-host-tools python"
+makedepends="karchive-devel kcodecs-devel ki18n-devel kdbusaddons-devel knewstuff-devel
+ kconfig-devel kiconthemes-devel kjobwidgets-devel kpimtextedit-devel akonadi-contacts-devel
+ kxmlgui-devel kservice-devel kdesignerplugin-devel kimap-devel libkdepim-devel"
+short_desc="Common libraries for KDE PIM"
+maintainer="John <johnz@posteo.net>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/pimcommon/template]: not valid SPDX identifier

**Comment by maxice8** [comment]
thank you for your efforts adopting the SPDX.

**Comment by Johnnynator** [comment]
It was on my todo list to fix the remaining Licenses (but I probably would have missed that GPL is also a decimal) (after having it compiled on Musl).
It would have been enough just to say that I should recheck them instead of commenting on every template.

**Comment by maxice8** [comment]
> It was on my todo list to fix the remaining Licenses (but I probably
would have missed that GPL is also a decimal) (after having it compiled on
Musl).
It would have been enough just to say that I should recheck them instead of
commenting on every template.

I personally prefer explicitly telling where the mistakes are but I'll keep
that in mind for when I next review a PR from you.

**Comment by Johnnynator** [comment]
Licenses should be fine now (correct SPDX and comma seprated)

**Comment by maxice8** [comment]
> I don't have a problem if you explicitly point to a mistake as long as it is not nearly 30 times the same one :D

I personally prefer explicitly telling where the mistakes are but I'll keep that in mind for when I next review a PR from you.

**Comment by Gottox** [comment]
thanks for this huge effort!

**Pull request merged**